elastic/apm-server
# Apply shutdown timeout to http server to limit reload delay

## Motivation/summary


httpServer.stop may block indefinitely, potentially due to misbehaving connections.

Apply shutdown timeout to http server so that in a hot reload, the old and new server overlapping time is bounded.

## Checklist




For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes



May need to change the code to hang the http server to observe the shutdown timeout. Alternatively there may be a way to mock a misbehaving http client without code change.

## Related issues


Part of #14337
